File: app/src/main/java/org/example/example/fabric/GameFabricator.java
package org.example.example.fabric;

import org.example.example.game.Board;
import org.example.example.game.Game;
import org.example.example.game.Player;
import org.example.example.game.WinCondition;
import org.example.example.piece.Color;
import org.example.example.piece.Piece;

import java.util.ArrayList;
import java.util.List;

public class GameFabricator {

    public static Game createClassicGame(){
        Board board = BoardFabricator.createNormalBoard();
        List<Piece> winConditionPieces = new ArrayList<>();
        winConditionPieces.add(board.getCordinate(1,5).getPiece());
        winConditionPieces.add(board.getCordinate(8,5).getPiece());
        WinCondition winCondition = new WinCondition(winConditionPieces);
        Player player1 = new Player(Color.WHITE, "white");
        Player player2 = new Player(Color.BLACK, "black");
        return new Game(player1 , player2, board, winCondition, board.getCordinate(1,4).getPiece());
    }
}

File: app/src/main/java/org/example/fabric/GameFabricator.kt
package org.example.fabric

import org.example.game.AntiChessGame
import org.example.game.Game
import org.example.game.WinCondition
import org.example.game.piece.Color

class GameFabricator {

    fun createClassicGame() : Game{
        val pieceFabricator = PieceFabricator()
        val boardFabricator = BoardFabricator()
        val board = boardFabricator.createNormalBoard()
        val winCondition = WinCondition(listOf(pieceFabricator.createKing(5, Color.WHITE) , pieceFabricator.createKing(29 , Color.BLACK)))
        return Game(board, winCondition, Color.WHITE, pieceFabricator.createQueen(4, Color.WHITE))
    }

    fun createAntiChessGame() : AntiChessGame{
        val pieceFabricator = PieceFabricator()
        val boardFabricator = BoardFabricator()
        val board = boardFabricator.createAntiChessBoard()
        val winCondition = WinCondition(listOf())
        return AntiChessGame(board, winCondition, Color.WHITE, pieceFabricator.createQueen(4, Color.WHITE))
    }
}